src/Testcontainers/Images/FileSystem.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace DotNet.Testcontainers.Images
+{
+  using System.Collections.Generic;
+  using System.IO;
+  using System.Linq;
+
+  public class FileSystem : IFileSystem
```

Comment:
The DockerIgnoreFile class didn't have any tests. The DockereIgnoreFile class uses File.ReadAllLines() and DirectoryInfo to access the filesystem. In order to test any changes you need to either write some temporary files and read those back or abstract the filesystem away and substitute it. I chose to do the latter. Do you prefer a different approach?

Reply:
> The DockerIgnoreFile class didn't have any tests.

It is implicitly covered by `ImageFromDockerfileTest`.

> I chose to do the latter. Do you prefer a different approach?

Not sure. I like the idea, but otherwise I would like to keep the changes minimal, and avoid unnecessary layers - I mean we are just reading a file. Do you think we can add a `.dockerignore` for each case to the assets and read the files from there (something like `accept_root_docker_files.dockerignore`, `deny_nested_docker_files.dockerignore` maybe you have better names)?